tests/python/test_bit_operations.py

```diff
@@ -0,0 +1,19 @@
+import taichi as ti
+import numpy as np
+import operator as ops
+from taichi import allclose
+import pytest
+
+func_table = [
+    (ops.lshift, ) * 2,
+]
+
+
+@pytest.mark.parametrize('ti_func,np_func', func_table)
+def test_python_scope_vector_binary_bit_operations(ti_func, np_func):
+    x = ti.Vector([2, 3])
+    y = ti.Vector([5, 4])
+
+    result = ti_func(x, y).to_numpy()
+    expected = np_func(x.to_numpy(), y.to_numpy())
+    assert allclose(result, expected)
```

Comment:
I guess the real test we need here, is something like
```python
@ti.test()
def test_shl():

    @ti.kernel
    def shl(a: ti.i32, b: ti.i32) -> ti.i32:
        return a << b
     
    assert shl(3, 0) == 3
    assert shl(3, 1) == 6
    assert shl(3, 2) == 12
   
    ....
```

Reply:
Yes, we need to test in Taichi-scope.

> enumerating all possible combinations

I don't think `enumerating all possible combinations` is a bad idea. It eliminates all possible hidden errors.

I agree with `it would be good to have smaller and simpler tests`. Had a hard time debugging tests when writing OpenGL.
How about to left a systematical test of `ops.lshift` in `test_element_wise.py`. and left a simpler one that only tests for scalar << scalar in this file like `/` and `%`'s scalar test does?
We can make test values in `test_element_wise.py` minimal, and values in `test_bit_operations.py` and `test_div.py` extraordinary for unit test.

Reply:
> I don't think `enumerating all possible combinations` is a bad idea. It eliminates all possible hidden errors.

If testing two components need an enumeration of all possible combinations, that probably means the designed systems are not very testable. We should try to avoid designs like that.